docs/design/fabric-v2+/interfaces.md

```diff
@@ -304,58 +241,117 @@ public int ecall_next_block(
-```c++
-// returns a cryptographic hash of the genesis block this tlcc enclave is initialized with.
-public hash get_channel_hash();
 
 // TODO update UMLs: these functions replace "check_ercc_commited_data"
 
-// verify state data
-public metadata get_state_metadata(
-        const char *namespace,
-        const char *key);
+// - verify state data
+
+//    public metadata get_state_metadata(
+//            const char *namespace,
+//            const char *key);
+message GetMetadataRequest {
+        string namespace   = 1;
+        string key         = 2;
+}
+message GetMetadataResponse {
+        bytes hash         = 1; 
+        // Note:
+        // - in first FPC implementation, we CMACed the hash; this authentication is done by the transparently by the session
+        // - encoding is SHA-256 over value found by key (or all-zero if key absent)
+}
+
+
+
+//    public metadata get_multi_state_metadata(
+//            const char *namespace,
+//            const char *comp_key);
+message GetMultiMetadataRequest {
+        string namespace   = 1;
+        string compo_key   = 2;
+}
+message GetMultiMetadataResponse {
+        bytes hash         = 1; // SHA-256 over value found by key (or all-zero if key absent)
+        // Note:
+        // - in first FPC implementation, we CMACed the hash; this authentication is done by the transparently by the session
+        // - encoding is SHA-256 over concatentation of key & values for all found keys and their values (or all-zero if no key absent)
+        //   TODO: above encoding is what we do now, but is malleable, so should be improved.
+}
+
 
-public metadata get_multi_state_metadata(
-        const char *namespace,
-        const char *comp_key);
+// - verify identities
 
 // verify that a given identity is part of a msp
 // the input is a serialized identity proto message as defined in
 // https://github.com/hyperledger/fabric-protos/blob/master/msp/identities.proto#L15
-public bool validate_identity(
-        const uint8_t *serializedIdentity,
-        const uint32_t len);
+//    public bool validate_identity(
+//            const uint8_t *serializedIdentity,
+//            const uint32_t len);
+message ValidateIdentityRequest {
+    bytes serializedIdentity = 1;
+}
+message ValidateIdentityResponse {
+    bool is_valid = 1;
+}
 
 // checks if a given enclave identifier can endorse transactions
 // as defined in the chaincode definition; this checks that the given enclave
 // has correct the MRENCLAVE and enclave is part of an organization that can
 // satisfy the endorsing policy of a given chaincode.
-public bool can_endorse(
-        const char *chaincode_id,       
-        const char *enclave_id);
+//    public bool can_endorse(
+//            const char *chaincode_id,
+//            const char *enclave_id);
+message CanEndorseRequest {
+    // note: both paramters can be implied from session context, so do not have to be transmitted
+}
```

Comment:
I believe also ERCC may want to query `can_endorse`, is this also possible without a secure channel between ercc and tlcc?

Reply:
i thought we solved that via lscc? (doesn't really need tlcc for it?)   That said, nothing prevents us to reuse the same messages and offer it in cleartext.  

Your question also made me realize, that i was actually wrong in my comment and `enclave_id` _cannot_ be inferred from context as the whole point of ecc calling this is to check it for _another_, so obviously the corresponding `enclave_id` has to be explicitly passed.   For you to reuse with ERCC, you would need though also need to explicitly add the `chaincode_id`.  This would be (a) redunandant for ecc and (b) raises the question whether ecc should be allowed to call this function for other chaincode.  The answer for the latter is probably 'info is public anyway, so shouldn't matter and for the former (redundancy) probably is who cares => will change correspondingly ...